runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -8058,8 +8058,57 @@ done:;
 	VMINLINE VM_BytecodeAction
 	defaultvalue(REGISTER_ARGS_LIST)
 	{
+retry:
+		j9object_t newObject = NULL;
 		VM_BytecodeAction rc = EXECUTE_BYTECODE;
-		// TODO: Implement bytecode
+		const U_16 index = *(U_16*)(_pc + 1);
+		J9ConstantPool * const ramConstantPool = J9_CP_FROM_METHOD(_literals);
+		J9RAMClassRef * const ramCPEntry = ((J9RAMClassRef*)ramConstantPool) + index;
+		J9Class * volatile resolvedClass = ramCPEntry->value;
+		bool doInitializeClass = FALSE;
+
+		if ((NULL != resolvedClass) && J9_IS_J9CLASS_VALUETYPE(resolvedClass)) {
+			if (!VM_VMHelpers::classRequiresInitialization(_currentThread, resolvedClass)) {
+				j9object_t instance = _objectAllocate.inlineAllocateObject(_currentThread, resolvedClass);
+				if (NULL == instance) {
+					updateVMStruct(REGISTER_ARGS);
+					instance = _vm->memoryManagerFunctions->J9AllocateObject(_currentThread, resolvedClass, J9_GC_ALLOCATE_OBJECT_INSTRUMENTABLE);
+					VMStructHasBeenUpdated(REGISTER_ARGS);
+					if (J9_UNEXPECTED(NULL == instance)) {
+						rc = THROW_HEAP_OOM;
+						goto done;
+					}
+				}
+				newObject = instance;
+				goto done;
+			}
+			doInitializeClass = TRUE;
+		}
+
+		buildGenericSpecialStackFrame(REGISTER_ARGS, 0);
+		updateVMStruct(REGISTER_ARGS);
+		if (doInitializeClass) {
+			/* Class requires initialization */
+			initializeClass(_currentThread, resolvedClass);
+		} else {
+			/* Class is unresolved or is not a value type */
+			resolveClassRef(_currentThread, ramConstantPool, index, J9_RESOLVE_FLAG_RUNTIME_RESOLVE | J9_RESOLVE_FLAG_INIT_CLASS | J9_RESOLVE_FLAG_INSTANTIABLE | J9_RESOLVE_FLAG_CHECK_VALUE_CLASS);
```

Comment:
Is there going to be another use of the CHECK_VALUE_CLASS flag? A failed resolution here will prevent any future resolution of the class ref.

Reply:
Actually, VT failures will not mark the CP entry (which may or may not be the correct behaviour). Still not convinced that we shouldn't just do the VT check here, and not pollute resolution with it.

Reply:
See above.